### DISCUSSION TITLE
Ecosystem specific utilities and code organization

### DISCUSSION THREAD

**Discussion by brandonjpierce** [gohypergiant/standard-toolkit#9] [General]
We have a high level notion of "validators", "converters", and "formatters" which usually just deals with primitive values.

We have an interesting question at hand of how do we want to organize things that may pertain to a particular ecosystem e.g. Apache Arrow?

For example, say we wanted to create a validator that checks to see if a particular GeoArrow geometry field is of a "Point" type. Would that go into a dedicated `@hypergiant/arrow` package? Would it go into our `@hypergiant/validators` package?

**Comment by belsrc** [comment]
As it can be broad, the issue was left open for the time being. General consensus at the moment is that we take a specificity approach. Using `validators` as an example, the general `validators` are at the language level. `arrow/validators` are at a technology level, so they are use items from `validators` but not the other way. `deckgl/validators` are at the library level, so they can use things from `arrow/validators` and `validators` but, again, not the other way.
